client/extensions/woocommerce/woocommerce-services/state/shipping-label/reducer.js

```diff
@@ -1153,6 +1154,21 @@ reducers[ WOOCOMMERCE_SERVICES_SHIPPING_LABEL_CLOSE_REFUND_DIALOG ] = state => {
 	};
 };
 
+reducers[ WOOCOMMERCE_SERVICES_SHIPPING_LABEL_STATUS_RETRIEVAL_IN_PROGRESS ] = ( state, { labelId } ) => {
+	const labelIndex = findIndex( state.labels, { label_id: labelId } );
+	const labelData = {
+		...state.labels[ labelIndex ],
+		statusRetrivalInProgress: true,
+	};
+
+	const newState = {
+		...state,
+		labels: [ ...state.labels ],
+	};
+	newState.labels[ labelIndex ] = labelData;
+	return newState;
+};
+
```

Comment:
Sorry, I have more questions! Do we want to have tests for reducers? They are all pure functions and probably don't need mocks.

Reply:
I don't know :) Test as often as possible but choose your tests wisely. Let's say that we want more tests - is this a good place to tests or we should pick something different - maybe better ( in the context of wcs ). What do you think?

Reply:
Yea, there are costs for us to achieve quality for our customers. So I agree to pick a good place and not spend time to add test simply for coverage. I don't know what the good place is though. 

Tests are most valuable when added in area we lack confidence in. If we don't need confidence in this area, then yea we don't need to test.